test/e2e/operator_test.go

```diff
@@ -1291,6 +1301,74 @@ func TestInternalLoadBalancerGlobalAccessGCP(t *testing.T) {
 	}
 }
 
+// TestAWSResourceTagsChanged tests the functionality of updating AWS resource tags
+// in the infrastructure configuration and validates that the expected
+// awsLBAdditionalResourceTags is set correctly on the
+// loadBalancer service associated with the default Ingress Controller.
+func TestAWSResourceTagsChanged(t *testing.T) {
+	t.Parallel()
+
+	if infraConfig.Status.Platform != "AWS" {
+		t.Skipf("test skipped on platform %q", infraConfig.Status.Platform)
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Second, defaultName, defaultAvailableConditions...); err != nil {
+		t.Errorf("did not get expected conditions: %v", err)
+	}
+
+	// Update AWS ResourceTags at infraConfig
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		infra, err := configClient.ConfigV1().Infrastructures().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if infra.Status.PlatformStatus == nil {
+			infra.Status.PlatformStatus = &configv1.PlatformStatus{}
+		}
+		if infra.Status.PlatformStatus.AWS == nil {
+			infra.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{}
+		}
+
+		infra.Status.PlatformStatus.AWS.ResourceTags = []configv1.AWSResourceTag{
+			{Key: "Key1", Value: "Value1"},
+			{Key: "Key2", Value: "Value2"},
+		}
+		_, updateErr := configClient.ConfigV1().Infrastructures().UpdateStatus(context.TODO(), infra, metav1.UpdateOptions{})
+		return updateErr
+	})
+
+	if retryErr != nil {
+		t.Errorf("failed to update infrastructure status: %v", retryErr)
+	}
+
+	awsLBAdditionalResourceTags := "service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags"
+	expectedTags := "Key1=Value1,Key2=Value2"
+	defaultIC := &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultName.Namespace,
+			Name:      defaultName.Name,
+		},
+	}
+	// validate awsLBAdditionalResourceTags for load balancer service of default ingress controller.
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		service := &corev1.Service{}
+		if err := kclient.Get(context.Background(), controller.LoadBalancerServiceName(defaultIC), service); err != nil {
+			t.Logf("failed to get service %s: %v", controller.LoadBalancerServiceName(defaultIC), err)
+			return false, nil
+		}
+		if actualTags, ok := service.Annotations[awsLBAdditionalResourceTags]; !ok {
+			t.Logf("load balancer has no %q annotation: %v", awsLBAdditionalResourceTags, service.Annotations)
+			return false, nil
+		} else if actualTags != expectedTags {
+			t.Logf("expected %s, found %s", expectedTags, actualTags)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("timed out waiting for the %s annotation to be updated: %v", awsLBAdditionalResourceTags, err)
+	}
```

Comment:
Can you use `assertServiceAnnotation` here?

Reply:
`assertServiceAnnotation` does not use `wait.PollImmediate`, and I think we should wait for the controller to update the annotation using the polling loop. I checked other tests like `TestInternalLoadBalancer`  and `TestAWSLBTypeChange` which are also using similar logic. If it's safe to use `assertServiceAnnotation` here, then we can update it.

Reply:
As we discussed on the call, I'm fine with not using `assertServiceAnnotation`.  You could add a code comment explaining why you cannot use `assertServiceAnnotation`, but I'm fine with the test in any case.

Reply:
Sure, I've added the following code comment explaining the reason
```go
	// Use a polling loop instead of assertServiceAnnotation since
	// the operator might not update the annotation immediately.
```